runtime/compiler/codegen/J9CodeGenerator.cpp

```diff
@@ -251,6 +251,48 @@ J9::CodeGenerator::fastpathAcmpHelper(TR::Node *node, TR::TreeTop *tt, const boo
    // next treetop and then at the current one
    TR::Block* prevBlock = tt->getEnclosingBlock();
    TR::Block* targetBlock = prevBlock->splitPostGRA(tt->getNextTreeTop(), cfg, true, NULL);
+
+   // Move call tree to end of block, along with any stores of the result of the call, if they're
```

Comment:
I wonder if the diagram on the above should be updated with `GlRegDeps()` to  show how the register dependency looks like between the split blocks and some sort of annotation that mentions the call tree needs to be at the end of the block.
https://github.com/eclipse/openj9/blob/bd3aac0b531b17e22787e8f8cf484522859c8f3f/runtime/compiler/codegen/J9CodeGenerator.cpp#L197-L219

Reply:
Thanks for suggesting that this should show the register dependencies; I'll do that.

As for showing that the call tree has to appear at the end of the block, the difficulty is that this diagram shows where the `acmphelper` call ends up after the call to `split`.  It's only at the end of the previous block just before the call `split`, so I would have to add another diagram showing the intermediate state.  I'm not sure it's worth it to add another diagram.  Thoughts?

Reply:
IMO a detailed explanation here before moving the treetops should suffice the intermediate state. Adding CFG diagram for intermediate state in the beginning would complicate.

Reply:
A detailed explanation would do.